Model/Product/Builder.php

```diff
@@ -135,15 +136,20 @@ public function __construct(
      * @param Product $product
      * @param Store $store
      * @param string $nostoScope
-     * @return NostoProduct
+     * @return NostoProduct|null
      */
     public function build(
         Product $product,
         Store $store,
         $nostoScope = self::NOSTO_SCOPE_API
     ) {
         $nostoProduct = new NostoProduct();
+        $modelFilter = new ModelFilter();
 
+        $this->eventManager->dispatch(
+            'nosto_product_load_before',
+            ['product' => $nostoProduct, 'magentoProduct' => $product, 'modelFilter' => $modelFilter]
+        );
```

Comment:
Check also here if `$modelFilter->isValid()` and if it returns false the builder should return null right away. No need to build Nosto product if it's going to be filtered out